tests/unit/descriptor/test_dapp.py

```diff
@@ -220,3 +220,63 @@ def test_http_proxy_descriptor(
         test_utils.verify_error(error, e)
     else:
         test_utils.verify_error(error, None)
+
+
+@pytest.mark.parametrize(
+    "descriptor_dict, error, expected_priority",
+    (
+        (
+            {
+                "payloads": {"foo": {"runtime": "vm"}},
+                "nodes": {
+                    "db1": {
+                        "payload": "foo",
+                        "entrypoint": [],
+                    },
+                    "http": {
+                        "payload": "foo",
+                        "entrypoint": [],
+                        "depends_on": ["db1", "db2"],
+                    },
+                    "db2": {
+                        "payload": "foo",
+                        "entrypoint": [],
+                    },
+                },
+            },
+            None,
+            ["db2", "db1", "http"],
```

Comment:
yes, I know, and yes, I believe this could be implementation-dependent but I think this is unlikely to change unless they change the underlying algorithm -> I'd leave it be until it actually causes problems -> updating the test should be easy